### PR TITLE
Fix undefined hostname on device create

### PR DESCRIPTION
**Pull request by molly-moen**
# Type of change? 

- [x] Bug fix
- [ ] New feature
- [ ] Enhancement
- [ ] Breaking change (breaks backward compatibility)

# Description, Context, Motivation 
Hostname was undefined in connection string due to casing error.

**Checklist:**

- [x] All tests passed
- [x] The code follows the code style and conventions of this project
- [ ] The change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: src/services/models/iotHubManagerModels.js [hunk 1]
```diff
@@ -24,7 +24,7 @@ export const toDeviceModel = (device = {}) => {
     'c2DMessageCount': 'c2DMessageCount',
     'enabled': 'enabled',
     'lastStatusUpdated': 'lastStatusUpdated',
-    'iotHubHostName': 'iotHubHostName',
+    'ioTHubHostName': 'ioTHubHostName',
```
Comment (ppathan) [inline src/services/models/iotHubManagerModels.js]: Isn't this naming convention wrong? iotHubHostName is camel case no idea what ioTHubHostName is :)

Comment (molly-moen) [reply src/services/models/iotHubManagerModels.js]: ioTHubHostName is what our code will translate as "camel case" from IoTHubHostName, which is what is returned. We can map ioTHubHostName to iotHubHostName in this and use that as the variable name

Comment (ppathan) [reply src/services/models/iotHubManagerModels.js]: Yeah can we do that keep client side iotHubHostName?

File: src/services/models/iotHubManagerModels.js [hunk 2]
```diff
@@ -24,7 +24,7 @@ export const toDeviceModel = (device = {}) => {
     'c2DMessageCount': 'c2DMessageCount',
     'enabled': 'enabled',
     'lastStatusUpdated': 'lastStatusUpdated',
-    'ioTHubHostName': 'ioTHubHostName',
+    'ioTHubHostName': 'iotHubHostName',
```
Comment (dluc) [inline src/services/models/iotHubManagerModels.js]: seems strange that the 't' is capital case on the left, lower case on the right. Could we use the same format?

Comment (molly-moen) [reply src/services/models/iotHubManagerModels.js]: Here is what is happening:
From IotHubManager we get IoTHubHostName
The UI auto-camelcases everything, which turns into ioTHubHostName
we want it to be iotHubHostName in the code, so this maps ioTHubHostName coming from our request to iotHubHostName, what we want the variable to be called.

Comment (dluc) [reply src/services/models/iotHubManagerModels.js]: thanks! that would be a great comment to add future readers

**Pull request merged**